File: tads_lineares/c/pilha.c
#include <stdio.h>

#define MAXTAM 10

// Atributos
int array[MAXTAM];
int n;

// Construtor
void start(){ n = 0; }

// Método de Inserção
void inserir(int num){
    if(n >= MAXTAM){
        printf("Array Cheio");
    } else {
        array[n] = num;
        n++;
    }
}

// Método de Remoção
void remover(){
    if(n == 0) {
        printf("Array Vazio");
    } else {
        n--;
    }
}

// Método de Exibição
void imprimir(){
    printf("[ ");
    for(int i = 0; i < n; i++){
        printf("%d ", array[i]);
    }
    printf("]\n");
}

int main(){
    start();
    inserir(1);
    inserir(5);
    inserir(10);
    inserir(25);
    imprimir();
    remover();
    remover();
    remover();
    imprimir();
    return 0;
}

File: tads_lineares/java/Pilha.java
package tads_lineares.java;

public class Pilha {
    // Atributos
    private int[] array;
    private int n;

    // Construtor
    public Pilha(int tamanho) {
        array = new int[tamanho];
        n = 0;
    }

    // Método de Inserção
    public void inserir(int num) throws Exception {
        if (n >= array.length) {
            throw new Exception("Array Cheio");
        } else {
            array[n] = num;
            n++;
        }
    }

    // Método de Remoção
    public void remover() throws Exception {
        if (n == 0) {
            throw new Exception("Array Vazio");
        } else {
            n--;
        }
    }

    // Método de Exibição
    public void imprimir() {
        System.out.print("[ ");
        for (int i = 0; i < n; i++) {
            System.out.print(array[i] + " ");
        }
        System.out.print("]");
    }
}
